src/devices/Bmxx80/Bme680.cs

```diff
@@ -222,5 +222,19 @@ private double CompensatePressure(int adcPressure)
 
             return calculatedPressure;
         }
+
+        /// <inheritdoc cref="Bmxx80Base"/>
+        protected override Temperature CompensateTemperature(int adcTemperature)
+        {
+            // Calculation for the Bme680 differs slightly from other sensors of this family!
+            var var1 = (adcTemperature / 16384.0 - _calibrationData.DigT1 / 1024.0) * _calibrationData.DigT2;
+            var var2 = adcTemperature / 131072.0 - _calibrationData.DigT1 / 8192.0;
+            var2 *= var2 * _calibrationData.DigT3 * 16.0;
```

Comment:
Seems like the only difference between Bme680 and the base implementation is the factor for multiplying this. If so, can we instead change it so that we have a protected virtual method that returns this factor method so that we only override that part? That way we don't have much duplicated code.

Reply:
Sure, I could just add a field to the base class (e.g. _tempCalibrationFactor) that bme680 can set to 16 and is 1 for the other ones. Would that be fine? Or should it really be a method that only returns this factor?

Reply:
a virtual getter method would take advantage of the polymorphic features of c#. I was thinking something super simple like this:

```c#
// on the base class, you have a virtual method like the following:
protected virtual double _factor => 1.0;

protected Temperature CompensateTemperature(int adcTemperature)
{
//...
    var2 *= var2 * _calibrationData.DigT3 * _factor;
//...
}



// And in your Bme680 subclass you can just do:
protected override double _factor => 16.0;
```

Reply:
also, now that I wrote it I just realized that you are doing `var2 *= var2 * ....`  Is that intended? do you actually want var2 squared? the previwous formula would expand to: `var2 = var2 * var2 * _calibrationData.DigT3 * 16.0;`

Reply:
> also, now that I wrote it I just realized that you are doing `var2 *= var2 * ....` Is that intended? do you actually want var2 squared? the previwous formula would expand to: `var2 = var2 * var2 * _calibrationData.DigT3 * 16.0;`

Yeah that's correct, var2 has to be squared.